homeassistant/components/neato/config_flow.py

```diff
@@ -0,0 +1,114 @@
+    """Neato integration config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize flow."""
+        self._username = vol.UNDEFINED
+        self._password = vol.UNDEFINED
+        self._vendor = vol.UNDEFINED
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+
+        if user_input is not None:
+            self._username = user_input["username"]
+            self._password = user_input["password"]
+            self._vendor = user_input["vendor"]
+
+            error = await self.hass.async_add_executor_job(
+                self.try_login, self._username, self._password, self._vendor
+            )
+            if error:
+                errors["base"] = error
+            else:
+                return self.async_create_entry(
+                    title=user_input[CONF_USERNAME],
+                    data=user_input,
+                    description_placeholders={"docs_url": DOCS_URL},
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                    vol.Optional(CONF_VENDOR, default="neato"): vol.In(VALID_VENDORS),
+                }
+            ),
+            description_placeholders={"docs_url": DOCS_URL},
+            errors=errors,
+        )
+
+    async def async_step_import(self, user_input):
+        """Import a config flow from configuration."""
+
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+        vendor = user_input[CONF_VENDOR]
+
+        error = await self.hass.async_add_executor_job(
+            self.try_login, username, password, vendor
+        )
+        if error is not None:
+            _LOGGER.error(error)
+            return self.async_abort(reason=error)
+
+        return self.async_create_entry(
+            title=f"{username} (from configuration)",
+            data={
+                CONF_USERNAME: username,
+                CONF_PASSWORD: password,
+                CONF_VENDOR: vendor,
+            },
+        )
+
+    @staticmethod
+    def try_login(username, password, vendor):
+        """Try logging in to device and return any errors."""
+        from pybotvac import Account, Neato, Vorwerk
+
+        this_vendor = None
+        if vendor == "vorwerk":
+            this_vendor = Vorwerk()
+        elif vendor == "neato":
+            this_vendor = Neato()
+        else:
+            return "invalid_vendor"
```

Comment:
Can we remove this case? Otherwise we should test it.

Reply:
No we need that since we use this method in __init__. I'll add the test

Reply:
The config schema validates this as well, right?

Reply:
Yea, you're right. Didn't thought about that. To be honest, I really don't like that if without an else. Anyway, the test is there now :tada:

Reply:
We could default to one of the vendors instead.

Reply:
I've changed that here: https://github.com/home-assistant/home-assistant/pull/26579/commits/4e30a56f3591e574fbc0d3dbe35686752f15f06a . But I really prefer the solution before https://github.com/home-assistant/home-assistant/pull/26579/commits/67e205d4bd1e7a5519c9de054792ef10d4ad8d42